README.md

```diff
@@ -1 +1,103 @@
 # Axelar Relayer core libraries
+This repo provides building blocks for Axelar<>Blockchain Relayer
+
+## Relayer Architecture Overview
+```mermaid
+graph TD
+    subgraph "Relayer System"
+        subgraph "Amplifier to Blockchain Flow"
+            amp_sub[Amplifier Subscriber]
+            tasks_queue[Amplifier Tasks Queue]
+            bcx_ing[Blockchain Ingester]
+        end
+
+        subgraph "Blockchain to Amplifier Flow"
+            bcx_sub[Blockchain Subscriber]
+            events_queue[Amplifier Events Queue]
+            amp_ing[Amplifier Ingester]
+        end
+    end
+ 
+    %% External systems
+    amp_api[/"Amplifier REST API"\]
+    bcx["Blockchain"]
+
+    %% Amplifier to Blockchain flow
+    amp_api -->|amplifier tasks| amp_sub
+    amp_sub -->|pushes amplifier tasks| tasks_queue
+    tasks_queue -->|consumes amplifier tasks| bcx_ing
+    bcx_ing -->|transforms & includes amplifier tasks| bcx
+ 
+    %% Blockchain to Amplifier flow
+    bcx -->|blockchain events| bcx_sub
+    bcx_sub -->|transforms to amplifier events| events_queue
+    events_queue -->|consumes amplifier events| amp_ing
+    amp_ing -->|sends amplifier events| amp_api
+
+    %% Styling
+    classDef component fill:#d9d2e9,stroke:#674ea7,stroke-width:2px;
+    classDef queue fill:#ffe6cc,stroke:#d79b00,stroke-width:2px;
+    classDef api fill:#d5e8d4,stroke:#82b366,stroke-width:3px,color:#000,font-weight:bold,font-size:18px;
+    classDef blockchain fill:#dae8fc,stroke:#6c8ebf,stroke-width:3px,color:#000,font-weight:bold,font-size:18px;
+
+    class amp_sub,bcx_ing,bcx_sub,amp_ing component;
+    class tasks_queue,events_queue queue;
+    class amp_api api;
+    class bcx blockchain;
+
+    %% Styling
+    classDef component stroke:#333,stroke-width:2px;
+    classDef queue fill:#ff9,stroke:#333,stroke-width:2px;
+    classDef external fill:#bbf,stroke:#333,stroke-width:1px;
+ 
+    class amp_sub,bcx_ing,bcx_sub,amp_ing component;
+    class tasks_queue,events_queue queue;
+    class amp_api,bcx external;
+```
+
+## Bidirectional Flow Architecture
+
+The relayer establishes bidirectional communication between an Amplifier API and a blockchain. It consists of these main flows:
+
+### Amplifier to Blockchain Flow
+- **Amplifier Subscriber**: Subscribes to the Amplifier REST API and receives amplifier tasks then sends them to queue
+- **Amplifier Tasks Queue**: Stores amplifier tasks
+- **Blockchain Ingester**: Consumes tasks from the queue, transforms them to a compatible format, and includes them in the blockchain
+
+### Blockchain to Amplifier Flow
+- **Blockchain Subscriber**: Subscribes to blockchain events, transforms them into amplifier events and sends to queue 
+- **Amplifier Events Queue**: Stores amplifier events
+- **Amplifier Ingester**: Consumes events from the queue and sends them to the Amplifier API
+
+## Internal Relayer Architecture
+
+The relayer uses threading and supervision model:
```

Comment:
@vaporif Can you add a paragraph for the new way we've split the processes, because the supervisor is not the main focus now?